mutation/mutation_rebuilder.hh

```diff
@@ -18,9 +18,10 @@ class mutation_rebuilder {
 public:
     explicit mutation_rebuilder(schema_ptr s) : _s(std::move(s)) { }
 
-    void consume_new_partition(const dht::decorated_key& dk) {
+    const mutation& consume_new_partition(const dht::decorated_key& dk) {
         assert(!_m);
         _m = mutation(_s, std::move(dk));
+        return *_m;
```

Comment:
Nitpick (but feel free to reject): This is a strange API - consume_new_partition returns to you a pointer, with unclear lifetime. Wouldn't it be clearer if consume_new_partition returned void (as before), and you had a separate getter to read from the current _m?

Reply:
For me it is implied that the mutation reference returned here is valid until `consume_end_of_partition()` is called.
I agree this might not be obvious.
A separate getter is problematic, because the builder doesn't always have a valid `mutation` object onboard. I then have to add checks or just "trust" that the getter is called in the right time. This way, you can only obtain a mutation reference when it is constructed internally, so you are guaranteed a valid reference. You just have to take care of not keeping it for too long.

Reply:
I will follow-up with a PR adding a comment about the lifecycle.

Reply:
https://github.com/scylladb/scylladb/pull/14853